L4.md

```diff
@@ -0,0 +1,77 @@
+more common (especially when a Java client library exists) to create a variant of the gem that declares as its target
+the `java` platform and expose a JAR-based implementation through a Ruby wrapper. This is the approach taken in other
+Ruby client libraries maintained by Google, e.g.
+[Google Protobuf](https://github.com/google/protobuf/blob/master/ruby/google-protobuf.gemspec#L12).
+
+
+### Related Proposals: 
+* Not Applicable
+
+## Proposal
+
+1. Use [JBundler](https://github.com/mkristian/jbundler) or a comparable tool to facilitate expressing a dependency on
+the Maven artifact published by the `gprc-java` project, exposing the JAR file to the Ruby build process when running on
+JRuby.
+1. Extend the existing [Gemspec](https://github.com/grpc/grpc/blob/master/grpc.gemspec) to add conditional logic that
+both includes the JAR file and any necessary wrapper code and excludes C implementation files when run on JRuby.
+1. Amend the standard gem release process to include publishing from the latest JRuby interpreter available at release
+time.
+
+## Rationale
+
+#### Using a native library vs creating a "pure" Ruby client
+A "pure" Ruby client with no dependency on a native library would work equally well on JRuby as other Ruby interpreters.
+It is assumed that the questions of whether the performance would suffer too greatly if it were not
+implemented in C and the maintenance burden of a completely standalone Ruby implementation are settled matters that were
+already discussed when the Ruby client was first implemented. Therefore this proposal does not suggest creating a "pure"
+Ruby implementation to address JRuby support.
+
+#### Creating a Maven dependency
+The current organization of GRPC's source code repositories creates a challenge that other projects that support a Ruby
+gem with dedicated support for clients running under JRuby do not have, namely that the Ruby source lives in a distinct
+project from the source of the Java client. This proposal suggests using Maven to solve the artifact dependency in order
+to avoid creating a requirement on a unified repository that includes the C, Ruby, and Java source in a single project.
+That approach works fine (this is how Google Protobuf handles the same problem), and might even be more optimal, but
+would require repository-level refactoring. The `grpc-java` client already exposes releases via Maven so there would be
+no work required on their part to support this proposal.
+
+## Implementation
+
+1. Update the Gemspec to add JBundler to the list of development dependencies when run on JRuby 
+1. Create a [JarFile](https://github.com/torquebox/maven-tools/wiki/Jarfile) to create a dependency on the grpc-java jar 
+1. Create wrapper code that delegates the existing Ruby API calls to the Java client, as needed
+1. Address any test regressions or expectation updates introduced by executing on JRuby
```

Comment:
We probably wouldn't be able to drop the jruby target into the current tests without changes to the tests themselves. This is probably doable but somewhat involved.

The testing suite would have to be updated to add "jruby", as basically a separate language, to the continuously tested language-platform matrix. Also I suspect a lot of the current grpc-ruby tests would have to be refactored to be agnostic of the C-wrapping and java-wrapping implementations (higher level tests should work find but there's lower level stuff in C-extensions is used in tests).